src/clustering_impl.h

```diff
@@ -121,13 +117,9 @@ class ConvexClustering {
   }
 
   bool admm_converged(){
-    return ((V - V_old).squaredNorm() + (Z - Z_old).squaredNorm() < 1e-7);
+    return (scaled_squared_norm(V - V_old) + scaled_squared_norm(Z - Z_old) < EXACT_STOP_PRECISION);
```

Comment:
Dumb question: Is there are reason we aren't checking `U - U_old` here? I can't think of one...

Reply:
No reason not to I suppose. My Rice email is valid for another year or so, which I imagine will be as long as I'm relevant to the project

Reply:
Wait. There is no U_old in clustering, which explains why it's omitted here

Reply:
Oh - interesting... I guess that was a memory saving thing at some point. I suppose it's harmless - at some point maybe I'll play around and see if it's worth dropping `U_old` from biclustering as well then.